### PR TITLE
Feature 46 initial fleet management

### LINKED ISSUE #46: Initial Fleet Management

### PR DESCRIPTION
**Pull request by villejuutila**
Resolves #46

### CODE REVIEW

File: src/components/dialogs/manage-device-dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    }
+    setLoading(false);
+  };
+
+  /**
+   * Event handler for text field change
+   */
+  const handleTextFieldChange = ({
+    target: { name, value }
+  }: React.ChangeEvent<HTMLInputElement>) =>
```
Comment (Jdallos) [inline src/components/dialogs/manage-device-dialog.tsx:48]: Destructure imports

File: src/components/dialogs/manage-device-dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    try {
+      await onConfirm(tempDevice);
+    } catch (error: any) {
+      console.error(error);
+    }
+    setLoading(false);
+  };
+
+  /**
+   * Event handler for text field change
```
Comment (Jdallos) [inline src/components/dialogs/manage-device-dialog.tsx:44]: Add missing @param docs

File: src/components/dialogs/manage-device-dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+   */
+  const handleTextFieldChange = ({
+    target: { name, value }
+  }: React.ChangeEvent<HTMLInputElement>) =>
+    setTempDevice({ ...tempDevice, [name as keyof typeof device]: value });
+
+  /**
+   * Event handler for switch change
```
Comment (Jdallos) [inline src/components/dialogs/manage-device-dialog.tsx:52]: Missing @param docs

File: src/components/dialogs/manage-device-dialog.tsx [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+   * Renders select box with given label, value and name
+   *
+   * @param children children
+   * @param value value
+   * @param name name
+   * @param label label
+   */
+  const renderSelect = (
+    name: string,
+    children: React.ReactNode,
```
Comment (Jdallos) [inline src/components/dialogs/manage-device-dialog.tsx:91]: Destructure imports

File: src/components/dialogs/manage-device-dialog.tsx [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+  /**
+   * Renders device models select box menu items
+   */
+  const renderDeviceModels = () =>
+    deviceModels.map((deviceModel) => (
+      <MenuItem key={deviceModel.id} value={deviceModel.id}>
+        {deviceModel.model}
+      </MenuItem>
+    ));
```
Comment (Jdallos) [inline src/components/dialogs/manage-device-dialog.tsx:117]:
```suggestion
    ));
    
```

File: src/components/floor-plan/floor-plan-info.tsx [hunk 6]
```diff
@@ -83,15 +88,36 @@ class FloorPlanInfo extends React.Component<Props, State> {
...
+      manageDeviceDialogOpen: false
     };
   }
 
   /**
    * Component render method
    */
   public render = () => {
-    return <div style={{ padding: theme.spacing(2) }}>{this.renderProperties()}</div>;
+    return (
```
Comment (Jdallos) [inline src/components/floor-plan/floor-plan-info.tsx:100]: Could this be an implicit return?

**Review by Jdallos — APPROVED** [review]

File: src/components/dialogs/manage-device-dialog.tsx (hunk 1 above)
Comment (villejuutila) [reply src/components/dialogs/manage-device-dialog.tsx:48]: Resolved

File: src/components/dialogs/manage-device-dialog.tsx (hunk 2 above)
Comment (villejuutila) [reply src/components/dialogs/manage-device-dialog.tsx:44]: Resolved

File: src/components/dialogs/manage-device-dialog.tsx (hunk 3 above)
Comment (villejuutila) [reply src/components/dialogs/manage-device-dialog.tsx:52]: Resolved

File: src/components/dialogs/manage-device-dialog.tsx (hunk 4 above)
Comment (villejuutila) [reply src/components/dialogs/manage-device-dialog.tsx:91]: Resolved

File: src/components/dialogs/manage-device-dialog.tsx (hunk 5 above)
Comment (villejuutila) [reply src/components/dialogs/manage-device-dialog.tsx:117]: Resolved

File: src/components/floor-plan/floor-plan-info.tsx (hunk 6 above)
Comment (villejuutila) [reply src/components/floor-plan/floor-plan-info.tsx:100]: Resolved

**Pull request merged**